app/controllers/products_controller.rb

```diff
@@ -1,29 +1,58 @@
 class ProductsController < ApplicationController
-  before_action :set_product, only:[:show, :edit, :update, :destroy]
+ before_action :set_product, only:[:show, :edit, :update, :destroy]
   
   def index
+    @products = Product.all
   end
-  
-  def show
-  end
-  
+
+  def show; end
+
   def dashboard
+    @products = Product.all
   end
-  
+
   def new
+    @product = Product.new
   end
 
   def create
+    begin
+      product = Product.create!(product_params.except(:image))
+      product.image.attach(product_params[:image])
+      
+      redirect_to dashboard_path
+    
+    rescue => exception
```

Comment:
It seems that you are not using `exception` here. Also, you will capture StandardError and all of its subtypes by not specifying an exact type of error to rescue. I think here you should capture the error thrown by Active Record (RecordInvalid)